### ISSUE TITLE
Integrity goes from sha1 -> sha512 and vice versa on differenct PCs

### ISSUE THREAD

**Issue by XantreDev** [pnpm/pnpm#7419]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

8.12.1

### Which area(s) of pnpm are affected? (leave empty if unsure)

Lockfile

### Reproduction steps

Run `pnpm i`.
For now i cannot provide repro, because I cannot warranty that it will work different in another env

### Describe the Bug

On coworkers PCs every time when they using `pnpm i` it fixes to `sha512` - my PC always tries to resolve it as `sha1`
Coworker on Windows too
```diff
   /media-typer@0.3.0:
-    resolution: {integrity: sha512-dq+qelQ9akHpcOl/gUVRTxVIOkAJ1wR3QAvb4RsVjS8oVoFjDGTc679wJYmUmknUF5HwMLOgb5O+a3KxfWapPQ==}
+    resolution: {integrity: sha1-hxDXrwqmJvj/+hzgAWhUUmMlV0g=}
     engines: {node: '>= 0.6'}
```

### Expected Behavior

It use the same integrity on different PCs

### Which Node.js version are you using?

20.10.0

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by KSXGitHub** [comment]
What are the pnpm versions in these PCs?

**Comment by XantreGodlike** [comment]
The same version, node 18.12. But I've tried another node version, too, but it behaves the same

**Comment by Aghassi** [comment]
@XantreGodlike can you check on both machines what version of `npm` you have installed? @joel.jeske and I have found that `pnpm info` just invokes `npm info` under the hood (you can see this with `pnpm info --verbose`). This means that versions of `npm` can be different between machines leading to different responses from registries depending on what is supported in that `npm` version I believe.

CC @zkochan can you confirm that this would be the case also? Want to make sure what we are observing is intended and accurate.

**Comment by XantreGodlike** [comment]
Oh, it sounds persuasive. We actually migrated to workspaces and the problem is not happen now

**Comment by Aghassi** [comment]
Sadly we use workspaces and we still have this problem 😓

**Comment by PoomSmart** [comment]
Still happens these days, pnpm 9.15.0

**Comment by zkochan** [comment]
pnpm will save the integrity that is returned by the registry in the package metadata json file. If the integrity changes then it means different integrities are returned by the registry.

**Comment by PoomSmart** [comment]
@zkochan Which means something is wrong from the registry's side?

**Comment by iki** [comment]
@XantreDev @PoomSmart we had similar issue. Turned out, some devs used the default npmjs.com registry (uses sha512), while some had company private Azure registry (uses sha1) in `~/.npmrc` on their machines.

Solution: put info about registry to use in getting started docs. Alternatively, configure the reqistry in project `.npmrc`, if it doesn't require auth.

Additionally, if you build your own private packages and use them from the private registry in 2 separate environments (e.g. development company and customer ones), then you need to build packages only once (e.g. in dev company pipeline) and make them accesible for the second environment one via npm registry, because building them in each env separately would create them with different sha's.

**Comment by zkochan** [comment]
Closing for lack of a reproduction on a supported version.

The integrity in the lockfile is the one the registry serves for the package. There is no reproduction here. Please open a new issue with one if two machines still write different integrities.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
